flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/common/utils/DataSetConversionUtil.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.common.utils;
+
+import org.apache.flink.api.common.functions.MapFunction;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.operators.SingleInputUdfOperator;
+import org.apache.flink.api.java.operators.TwoInputUdfOperator;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.ml.common.MLEnvironment;
+import org.apache.flink.ml.common.MLEnvironmentFactory;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.types.Row;
+
+/**
+ * Provide functions of conversions between DataSet and Table.
+ */
+public class DataSetConversionUtil {
+	/**
+	 * Convert the given Table to {@link DataSet}<{@link Row}>.
+	 *
+	 * @param sessionId the sessionId of {@link MLEnvironmentFactory}
+	 * @param table the Table to convert.
+	 * @return the converted DataSet.
+	 */
+	public static DataSet <Row> fromTable(Long sessionId, Table table) {
+		return MLEnvironmentFactory
+			.get(sessionId)
+			.getBatchTableEnvironment()
+			.toDataSet(table, Row.class);
+	}
+
+	/**
+	 * Convert the given DataSet into a Table with specified TableSchema.
+	 *
+	 * @param sessionId the sessionId of {@link MLEnvironmentFactory}
+	 * @param data   the DataSet to convert.
+	 * @param schema the specified TableSchema.
+	 * @return the converted Table.
+	 */
+	public static Table toTable(Long sessionId, DataSet <Row> data, TableSchema schema) {
```

Comment:
This API seems to be not tested.

Reply:
also if `schema.getFieldTypes()` is deprecated. I would rather we remove this function and just allow the user to handle deprecation in user code. (this would be much cleaner for future maintenance)

Reply:
Thanks, The case has been added.
we must set the table using the column types (see below) and the TypeInformation is used in DataSet Api, so the `TableSchema` is necessary.

Reply:
I would add some comments here.